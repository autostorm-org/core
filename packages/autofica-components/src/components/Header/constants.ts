enum HeaderLocations {
  left,
  center,
  right,
}

export { HeaderLocations };
