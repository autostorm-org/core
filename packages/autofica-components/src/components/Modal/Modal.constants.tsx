const modalElementId = "__AUTOFICA_MODAL_CONTAINER";
const getModalErrorMessage = (modalElementId: string) =>
  `Unable to find modal or root element with  provided id modal: ${modalElementId}`;
export { modalElementId, getModalErrorMessage };
