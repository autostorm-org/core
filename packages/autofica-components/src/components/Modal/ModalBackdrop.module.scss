@import "../../theme/constants.scss";
@import "../../theme/z-indexes.scss";

.root {
  backdrop-filter: blur(2px); /* Chrome and Opera */
  background-color: rgb(0, 0, 0, 0.4);
  width: 100vw;
  height: 100vh;
  position: fixed;
  z-index: $modal-backdrop-z-index;

  top: 0px;
  right: 0px;
  left: 0px;
  bottom: 0px;
  // opacity: var(--modal-opacity);
}

.visibleRoot {
  display: block;
}

.invisibleRoot {
  display: none;
}
.contentContainer {
  display: inline-block;
  position: relative;
  transform: translateX(-50%);
  max-width: 90%;
  width: 500px;
  top: 10%;
  left: 50%;
  @media (min-width: $layout-tablet-min) {
    top: 15%;
  }
}
