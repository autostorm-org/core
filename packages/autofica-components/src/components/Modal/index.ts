//Modal
export { default as Modal } from "./Modal";
export type { IModalProps } from "./Modal";

//Modal constants
export { modalElementId, getModalErrorMessage } from "./Modal.constants";

//ModalBackdrop
export { default as ModalBackdrop } from "./ModalBackdrop";
export type { IModalBackdropProps } from "./ModalBackdrop";

//ModalDivider
export { default as ModalDivider } from "./ModalDivider";
export type { IModalHeaderProps } from "./ModalDivider";

//ModalPortal
export { default as ModalPortal } from "./ModalPortal";
export type { ModalPortalProps } from "./ModalPortal";
