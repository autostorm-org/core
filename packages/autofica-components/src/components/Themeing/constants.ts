const themeingClassName = "__AUTOFICA_THEME";

export { themeingClassName };
