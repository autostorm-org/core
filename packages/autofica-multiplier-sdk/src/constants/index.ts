enum WalletConnectionStates {
  NOT_CONNECTED,
  METAMASK_WALLET,
  BSC_WALLET,
  NO_WALLET_DETECTED,
}

enum SupportedNetworks {
  MAINNET = 56,
  TESTNET = 97,
}

const ChainData = {
  [SupportedNetworks.MAINNET]: {
    name: "BSC Mainnet",
  },

  [SupportedNetworks.TESTNET]: {
    name: "BSC Testnet",
  },
};

const Contracts = {
  [SupportedNetworks.MAINNET]: {
    lendingPoolAddressProvider: "0xCc0479a98cC66E85806D0Cd7970d6e07f77Fd633",
  },

  [SupportedNetworks.TESTNET]: {
    lendingPoolAddressProvider: "0x78547CBf195Dc3D92C5847acD9E89aFB35430c0f",
  },
};

export { WalletConnectionStates, SupportedNetworks, Contracts, ChainData };
