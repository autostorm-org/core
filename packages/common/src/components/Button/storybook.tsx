import React from "react";
import Button from "./Button";

// This default export determines where you story goes in the story list
export default {
  title: "Button",
  component: Button,
};

export const ButtonStory = () => {
  return <Button>Hello</Button>;
};
