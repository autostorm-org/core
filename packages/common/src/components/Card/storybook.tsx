import React from "react";
import Card from "./Card";
import Typography, { EnumTypographyVariant } from "../Typography";
import Button from "../Button";
import { withKnobs, boolean } from "@storybook/addon-knobs";
import { ThemeContext, nightTheme, defaultTheme } from "../../theme";
import ResponsiveTemplate from "../ResponsiveTemplate";
// This default export determines where you story goes in the story list
export default {
  title: "Card",
  component: Card,
  decorators: [withKnobs],
};

const MobileCard = () => {
  return (
    <div style={{ height: 184, width: 350 }}>
      <Card>
        <Typography variant={EnumTypographyVariant.h3}>Card title</Typography>
        <br />
        <Typography variant={EnumTypographyVariant.p}>
          Card body contents. Card body contents. Card body contents. Card body
          contents.
        </Typography>
        <br />
        <Button onClick={() => {}} overrides={{ marginTop: 9 }}>
          Call to action
        </Button>
      </Card>
    </div>
  );
};

const DesktopCard = () => {
  return (
    <div style={{ height: 240, width: 560 }}>
      <Card style={{ padding: "30px" }}>
        <Typography variant={EnumTypographyVariant.h1}>
          Crypto car loan finder
        </Typography>
        <br />
        <Typography variant={EnumTypographyVariant.p}>
          Get a collateralized loan for your next car purchase. Chose your
          backing assets. Powered by Multiplier.{" "}
        </Typography>
        <br />
        <Button onClick={() => {}} overrides={{ marginTop: 10 }}>
          Call to action
        </Button>
      </Card>
    </div>
  );
};

const ResponsiveStory = () => {
  const isDayMode = boolean("DAY MODE", true);
  return (
    <ThemeContext.Provider value={isDayMode ? defaultTheme : nightTheme}>
      <ResponsiveTemplate
        mobile={MobileCard}
        desktop={DesktopCard}
        tablet={DesktopCard}
      />
    </ThemeContext.Provider>
  );
};

export const FirstStory = ResponsiveStory.bind({});
