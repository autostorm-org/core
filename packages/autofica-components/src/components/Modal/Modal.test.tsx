import React, { useState } from "react";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";
import { ModalStory } from "./Modal.storybook";
import { modalElementId } from "./Modal.constants";

let container;

beforeEach(() => {
  container = document.createElement("aside");
  container.id = modalElementId;
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});
describe("Modal Test", () => {
  it("Should render Modal component", () => {
    act(() => {
      ReactDOM.render(<ModalStory />, container);
    });

    expect(container).toMatchSnapshot();
  });
});
