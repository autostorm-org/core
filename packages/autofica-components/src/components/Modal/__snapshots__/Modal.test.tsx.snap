// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Test Should render Modal component 1`] = `
<aside
  id="__AUTOFICA_MODAL_CONTAINER"
>
  <div
    class="root invisibleRoot"
  >
    <div
      class="contentContainer "
    >
      <div
        class="root card "
      >
        <div>
          <div>
            <button
              aria-label="Close Modal Button"
              class="root "
              style="margin-left: auto; display: block; position: absolute; top: 10px; right: 10px;"
            >
              <svg
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <line
                  x1="15"
                  x2="9"
                  y1="9"
                  y2="15"
                />
                <line
                  x1="9"
                  x2="15"
                  y1="9"
                  y2="15"
                />
              </svg>
            </button>
          </div>
          <span
            class="h3"
          >
            hellop
          </span>
          <div
            class="root"
          />
          <br />
          <span
            class="p"
          >
            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
          </span>
          <br />
          <button
            class="root "
            style="margin-top: 20px;"
          >
            Hello
          </button>
        </div>
      </div>
    </div>
  </div>
  <button
    class="root "
  >
    Open Modal
  </button>
</aside>
`;
